### ISSUE TITLE
Possible cause of the `empty String` error?

### ISSUE THREAD

**Issue by ncpa0cpl** [star-micronics/react-native-star-io10#130]
## Description
At my work we've encountered an issue like this: when calling the `StarPrinter.getStatus()` we'd get an `empty String` exception on one of the production devices, it happens only on one of them, and started happening after a network outage. Looking through the stack trace it appears that the SDK parses a string into a number, but the string happens to be empty hence the error. Though we can not think of what could be causing this, in the first place what numbers are there to be parsed in a getStatus() call?

Could someone shed some light on what could possible be the cause of this?

## Your device where the bug occurs 
 - Device: 
 - OS:
    android 12
 - Browser: 
 - Version:

## Your printer
- Model Names:
    Star mC-Print3
- Firmware Version:
- Interface:
    Ethernet Lan

SDK Version: 1.6.0

## Additional context

Here's the stack trace of the error that we are getting:

```json
{
		"0": {
			"file": "FloatingDecimal.java",
			"methodName": "readJavaFormatString",
			"lineNumber": 1842,
			"class": "sun.misc.FloatingDecimal"
		},
		"1": {
			"file": "FloatingDecimal.java",
			"methodName": "parseDouble",
			"lineNumber": 110,
			"class": "sun.misc.FloatingDecimal"
		},
		"2": {
			"file": "Double.java",
			"methodName": "parseDouble",
			"lineNumber": 538,
			"class": "java.lang.Double"
		},
		"3": {
			"file": "SourceFile",
			"methodName": "c",
			"lineNumber": 15,
			"class": "com.starmicronics.stario10.starxpandcommand.json.b"
		},
		"4": {
			"file": "SourceFile",
			"methodName": "f",
			"lineNumber": 53,
			"class": "com.starmicronics.stario10.starxpandcommand.json.b"
		},
		"5": {
			"file": "SourceFile",
			"methodName": "a",
			"lineNumber": 1,
			"class": "com.starmicronics.stario10.starxpandcommand.json.b"
		},
		"6": {
			"file": "SourceFile",
			"methodName": "b",
			"lineNumber": 1,
			"class": "com.starmicronics.stario10.autoswitchinterface.a$a"
		},
		"7": {
			"file": "SourceFile",
			"methodName": "a",
			"lineNumber": 3,
			"class": "com.starmicronics.stario10.autoswitchinterface.a$a"
		},
		"8": {
			"file": "SourceFile",
			"methodName": "a",
			"lineNumber": 12,
			"class": "com.starmicronics.stario10.StarPrinter"
		},
		"9": {
			"file": "SourceFile",
			"methodName": "a",
			"lineNumber": 19,
			"class": "com.starmicronics.stario10.StarPrinter"
		},
		"10": {
			"file": "SourceFile",
			"methodName": "b",
			"lineNumber": 2,
			"class": "com.starmicronics.stario10.StarPrinter"
		},
		"11": {
			"file": "SourceFile",
			"methodName": "access$internalOpen",
			"lineNumber": 1,
			"class": "com.starmicronics.stario10.StarPrinter"
		},
		"12": {
			"file": "SourceFile",
			"methodName": "invokeSuspend",
			"lineNumber": 77,
			"class": "com.starmicronics.stario10.StarPrinter$s"
		},
		"13": {
			"file": "ContinuationImpl.kt",
			"methodName": "resumeWith",
			"lineNumber": 8,
			"class": "hn.a"
		},
		"14": {
			"file": "DispatchedTask.kt",
			"methodName": "run",
			"lineNumber": 117,
			"class": "bo.s0"
		},
		"15": {
			"file": "CoroutineScheduler.kt",
			"methodName": "run",
			"lineNumber": 98,
			"class": "io.a$a"
		}
}
```

**Comment by PawelLampliapps** [comment]
It is also worth to mention that recycling power on the device and the printer doesn't help. Only after uninstalling the app and installing it again, we can establish connection with the printer.
I was looking through issues in this repo and in https://github.com/star-micronics/StarXpand-SDK-Android, and I didn't find anything similar except this issue: https://github.com/star-micronics/StarXpand-SDK-Android/issues/19 where reinstall was also the only solution. In first comment, it was told that it is possible that it is caused by "recycling stored values". Is there a way for clearing those values as a temporary fix?

**Comment by can-miki** [comment]
@ncpa0cpl 
Sorry for the late reply.
Please refer to the SDK manual for details about the items retrievable with [getStatus()](https://star-m.jp/products/s_print/sdk/react-native-star-io10/manual/en/api-reference/star-printer/get-status.html) [Return value: [StarPrinterStatus](https://star-m.jp/products/s_print/sdk/react-native-star-io10/manual/en/api-reference/star-printer-status/index.html#star-printer-status)].

I used [the sample for getStatus](https://github.com/star-micronics/react-native-star-io10/blob/master/example/samples/status/App.tsx) to verify the issue, but I could not reproduce the same problem.
Therefore, we would like to request additional information about your current situation. Could you please provide the following details?

1. Background information:
Was this issue occurring with a printer that had previously been connected successfully, or was this your first attempt at connecting the printer?
Have there been any recent changes to your library, device, or network configuration?

2. About your network environment (because this is Ethernet issue):
Is there any factor in your network that could be causing an obstacle (e.g., specific firewall settings or network restrictions)?

3. Bug report:
For reference, we would like to know the information about react-native info.



If you have any additional information, or questions, or need any assistance, please feel free to let us know.

**My testing environment:**
Device: Pixel 8a (Android)
OS: 15

Printer Model Names: MCP31CB
Printer Firmware Version: 3.8
Printer Interface: Lan (Ethernet)

react-native info:
> System:
> OS: macOS 14.6.1
> CPU: (8) arm64 Apple M2
> Memory: 420.17 MB / 8.00 GB
> Shell:
> version: "5.9"
> path: /bin/zsh
> Binaries:
> Node:
> version: 22.11.0
> path: /usr/local/bin/node
> Yarn:
> version: 1.22.22
> path: /usr/local/bin/yarn
> npm:
> version: 10.9.0
> path: /usr/local/bin/npm
> Watchman:
> version: 2024.07.01.00
> path: /usr/local/bin/watchman
> Managers:
> CocoaPods:
> version: 1.15.2
> path: /usr/local/bin/pod
> SDKs:
> iOS SDK:
> Platforms:
> - DriverKit 24.1
> - iOS 18.1
> - macOS 15.1
> - tvOS 18.1
> - visionOS 2.1
> - watchOS 11.1
> Android SDK: Not Found
> IDEs:
> Android Studio: 2024.1 AI-241.18034.62.2412.12266719
> Xcode:
> version: 16.1/16B40
> path: /usr/bin/xcodebuild
> Languages:
> Java:
> version: 17.0.11
> path: /usr/bin/javac
> Ruby:
> version: 2.6.10
> path: /usr/bin/ruby
> npmPackages:
> "@react-native-community/cli": Not Found
> react:
> installed: 18.3.1
> wanted: 18.3.1
> react-native:
> installed: 0.75.4
> wanted: 0.75.4
> react-native-macos: Not Found
> npmGlobalPackages:
> "react-native": Not Found
> Android:
> hermesEnabled: true
> newArchEnabled: false
> iOS:
> hermesEnabled: true
> newArchEnabled: false

**Comment by can-miki** [comment]
@PawelLampliapps 
Thank you for sharing your information/insight.

At this point, I could not determine whether this issue was directly related to the problem.
If possible, could you create a new issue for this?
When creating the issue, it would be helpful if you could share the reproduction steps again based on the bug report. (Please note that this issue is for React Native)

I appreciate your cooperation.

**Comment by ncpa0cpl** [comment]
> Was this issue occurring with a printer that had previously been connected successfully

yes, it was working fine for a long time before

> Have there been any recent changes to your library, device, or network configuration?

I think it was around the time when we upgraded star-io sdk from 1.4.0 to 1.6.0

> Is there any factor in your network that could be causing an obstacle (e.g., specific firewall settings or network restrictions)?

there shouldn't be, as I said it was working just fine until it somehow got into this bad state, we tried changing some network settings on the router as well as rebooting all the different devices in the chain but nothing helped. only app reinstall could fix the problem

we also were unable to reproduce this issue, it seems there has to be some specific condition that needs to be met and then it enters a bad state, though we have no idea what it is

here's my react-info (though the issue occurred on a build created on a CI pipeline so a lot if it is irrelevant):
```
System:
  OS: Linux 5.10 Manjaro Linux
  CPU: (8) x64 Intel(R) Core(TM) i7-10610U CPU @ 1.80GHz
  Memory: 19.61 GB / 30.99 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.18.2
    path: /tmp/yarn--1735814051546-0.19985730622597142/node
  Yarn:
    version: 1.22.22
    path: /tmp/yarn--1735814051546-0.19985730622597142/yarn
  npm:
    version: 9.8.1
    path: /usr/local/bin/npm
  Watchman: Not Found
SDKs:
  Android SDK: Not Found
IDEs:
  Android Studio: AI-241.18034.62.2411.12169540
Languages:
  Java: Not Found
  Ruby:
    version: 2.6.8
    path: /home/owner/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.74.5
    wanted: 0.74.5
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: false
  newArchEnabled: false
```

on the CI we use these:
Java 22
Gradle 8.7
Node v18.18.0
Android NDK 25.2.9519653
Android SDK 34.0.0

**Comment by can-miki** [comment]
@ncpa0cpl  I have a few questions to help me understand the problem.
You said that reinstalling the app will fix the problem, but will the problem come back sometime after reinstalling the app? How often does this error occur?
Could it have something to do with data cycling, as @PawelLampliapps mentioned?

If there's anything that could be the cause, please let me know.

**Comment by can-miki** [comment]
We apologize for the wait.

This problem did not occur during our testing evaluation.
If we could reproduce the problem here we would be able to investigate, but it is difficult to identify the problem since we couldn't reproduce it.
At this time, we do not know the conditions under which the problem can be reproduced, so it is difficult to isolate the problem (the environmental issue? the library issue?).  

The stack trace you provided has been obfuscated, so it will take time to analyze the code. We apologize for the inconvenience.

Additionally, please allow me to confirm the situation. The purpose is to confirm the urgency of this issue.

- Is this issue still occurring?
Does restarting the device solve the problem eternal? I'd like to know the current situation - how often this happens.

- What kind of specific problems do you have when an `empty string` occurs? Does the app freeze? Can't print?

- Does the latest version as of January 2025 (1.8.0 ) not resolve the issue?
So far, after looking at [the release history](https://github.com/star-micronics/react-native-star-io10/releases), we have not found any changes that could cause this issue.
> Have there been any recent changes to your library, device, or network configuration?
>  I think it was around the time when we upgraded star-io sdk from 1.4.0 to 1.6.0

**Comment by ncpa0cpl** [comment]
1. We encountered this error twice around the same time frame on two different devices at two different locations, to my knowledge not occurred again since (back then restating the device had no effect on the issue whatsoever)
2. inability to print or pull the printer status, otherwise the app was working fine
3. as said above, we have not run into this issue again, but it's hard to say for sure if it has been fixed or not

**Comment by can-miki** [comment]
@ncpa0cpl 
Thank you for your message.

Our R&D team is currently conducting an investigation to understand this error.

We will provide you with an update on the investigation status at the end of February.
Thank you for your continued support.

**Comment by can-miki** [comment]
The analysis results are as follows.
If you have any questions, please let me know. Thank you.

■ Solution  
Update stario10 to **Ver1.7.0 or later**.

■ Investigation Results  
Based on the analysis after restoring the stack trace, it is assumed that there is an issue with the `autoswitchinterface` process during the execution of `openAsync`.  
⇒ This issue has already been resolved in **stario10 Ver1.7.0 or later**.

Although the Issue mentions `getstatus()`, it is believed that the actual problem occurs during the execution of `openAsync()`.  
The reason why the problem occurred when the library was changed from **Ver1.4.0 to Ver1.6.0** is thought to be due to the difference in the default value of `autoswitchinterface` between **stario10 Ver1.4.0 (disabled)** and **Ver1.6.0 (enabled)**.

It is unclear if the reproduction method is the same as the user's, but there is a possibility that the same stack trace will occur if the device is powered off or the OS hangs immediately after executing `openAsync`.

**Comment by can-miki** [comment]
This has been answered and the issue will be closed. I appreciate your inquiry.
